plugins/node/opentelemetry-instrumentation-winston/package.json

```diff
@@ -58,12 +59,14 @@
     "sinon": "15.2.0",
     "test-all-versions": "6.0.0",
     "ts-mocha": "10.0.0",
-    "typescript": "4.4.4",
-    "winston": "3.3.3",
-    "winston2": "npm:winston@2.4.7"
+    "typescript": "4.4.4"
   },
   "dependencies": {
-    "@opentelemetry/instrumentation": "^0.46.0"
+    "@opentelemetry/api-logs": "^0.45.1",
+    "@opentelemetry/core": "1.18.1",
+    "@opentelemetry/instrumentation": "^0.46.0",
+    "winston": "3.3.3",
+    "winston2": "npm:winston@2.4.7"
```

Comment:
Having a runtime dependency on winston@2 would be unfortunate. Would it be possible to just have a peer dep instead? Most of the winston@2 era transports that I looked at do *not* have a direct dependency on `winston` listed in their package.json. (FWIW the don't list `winston` in `peerDependencies` either, mostly I think because winston@2 predated `peerDependencies` existing or being well-known in npm.)

This instrumentation having a runtime dep on `winston@2` means carrying along its transitive deps, and winston@2's last release was 2022-11-15. Those deps aren't *that* big (mostly it is ~4MB from lodash); and the deps don't currently have an security warnings.

If it is hard/impossible to make a winston@2 transport subclass without a direct dep on winston@2, a possible consideration is to only add log sending support for winston@3.

Reply:
Ok, I checked a couple of winston custom transports and basically they have winston as a peer dependency, and for winston <2 they basically only have a dev dependency, so they will fail if winston is not also installed in the app. Easy fix here is to add a peer dependency but like @Flarna mentioned in other comment this would cause several issues and does not align with current guidelines, I can explore suggested approach by @seemk but I can immediately see utils.inherits is a legacy API and discouraged in the official docs, so not sure how to proceed.

Reply:
> Ok, I checked a couple of winston custom transports and basically they have winston as a peer dependency, and for winston <2 they basically only have a dev dependency, so they will fail if winston is not also installed in the app. Easy fix here is to add a peer dependency but like @Flarna mentioned in other comment this would cause several issues and does not align with current guidelines, I can explore suggested approach by @seemk but I can immediately see utils.inherits is a legacy API and discouraged in the official docs, so not sure how to proceed.

While it is legacy in Node.js, the [implementation is quite simple](https://github.com/nodejs/node/blob/v21.6.1/lib/util.js#L241-L260), i.e. `Object.defineProperty` and `Object.setPrototypeOf`. Just an idea :slightly_smiling_face:

Reply:
I think this is resolved. IIRC when I added this comment the "winston2" dep was in "dependencies" rather than in "devDependencies" where it is now.